UI: Cleaning UI Backward Compatibility Code

ganimtron-10 opened the pull request:
As per out last discussion, we will be dropping the v1 UI backward compatibility as it introduces complexity and brings a high risk of error. This PR cleans that compatibility code.
Will add the migration guide along with the Panel2D PR.

ganimtron-10 committed:
fix: clean UI backward compatibility code

ganimtron-10 commented:
Tests are failing due to pygfx new release v0.14.0
Will try to fix those later today.

maharshi-gor commented:
Hi @ganimtron-10 

Thanks for the work. To give you a primer on the error you see in the CIs.

It is due to with `Pygfx: 0.14.0` has bumped up to `wgpupy:0.25` which has change the what will be returned as Canvas and methods for that has been changed.

We need to update the method for snapshot from `self.window.draw_frame()` to `self.window.draw()`

Ref: https://docs.pygfx.org/stable/_gallery/introductory/offscreen.html#sphx-glr-gallery-introductory-offscreen-py

Ref: https://github.com/pygfx/pygfx/releases/tag/v0.14.0
Ref: https://github.com/pygfx/wgpu-py/blob/main/CHANGELOG.md

Let me know if that fixes the problem.

maharshi-gor commented on fury/window.py:
```diff
@@ -566,18 +560,13 @@ def __init__(
-        use_old_ui : bool, optional
-            Whether to use old ui with anchor as LEFT, BOTTOM or new with LEFT, TOP.
         """
         self._size = size
```
Can you pass this size to the UI Context directly from here ?

ganimtron-10 replied:
do you mean using ```UIContext.canvas_size = size```  here

and removing the last ``` self._resize(self._size)```?

maharshi-gor reviewed (commented):
Hi @ganimtron-10 

Please check the comment on workflows and size change in UI Context

ganimtron-10 commented:
> Hi @ganimtron-10
> 
> Thanks for the work. To give you a primer on the error you see in the CIs.
> 
> It is due to with `Pygfx: 0.14.0` has bumped up to `wgpupy:0.25` which has change the what will be returned as Canvas and methods for that has been changed.
> 
> We need to update the method for snapshot from `self.window.draw_frame()` to `self.window.draw()`
> 
> Ref: https://docs.pygfx.org/stable/_gallery/introductory/offscreen.html#sphx-glr-gallery-introductory-offscreen-py
> 
> Ref: https://github.com/pygfx/pygfx/releases/tag/v0.14.0 Ref: https://github.com/pygfx/wgpu-py/blob/main/CHANGELOG.md
> 
> Let me know if that fixes the problem.

Hi @maharshi-gor,

Thanks for the context. I had resolved the draw_frame changes, but there are few more things which are impacted from this change and was trying to resolve those.

1. get_app was removed from wgpu.auto.qt. Error Details: https://github.com/fury-gl/fury/actions/runs/18495214933/job/52697737150#step:13:2198

2. wgpu.auto is deprecated we have to use rendercanvas. Run any example and you will see the warning in the console. ```The wgpu.gui.auto is deprecated, use rendercanvas.auto instead.
The wgpu.gui.offscreen is deprecated, use rendercanvas.offscreen instead.```

The below one I am not sure of is related to the same just adding it for info:
Getting this below log while executing some examples
```Unable to find extension: VK_EXT_physical_device_drm```

skoudoro reviewed (changes requested):
Why a context is still needed since we removed the backward compatibility ? 

Can you remove all deprecattion warnings also

Thanks

ganimtron-10 committed:
Merge branch 'v2' into v2_ui_clean

ganimtron-10 commented:
> Why a context is still needed since we removed the backward compatibility ?

UI Context wasnt added as a part of backward compatibility.
The main role of UI Context is to supply the current canvas size to the UI elements so they can position themselves correctly in Top Left fashion (y down) because the ScreenCoords follow Bottom Left format (y up).
Also later on we will use the `active` and `hot` UI from the UIContext to handle the hovering, dragging, keyboard shortcuts and all.

> Can you remove all deprecattion warnings also

I have removed all of the deprecation warnings

skoudoro reviewed (approved):
Thanks for the clarification @ganimtron-10 

The PR is good to go @maharshi-gor, feel free to merge it if you are ok

maharshi-gor reviewed (approved):
Thank you for the work @ganimtron-10 

LGTM! Merging!
